src/functions/negate.js

```diff
@@ -3,13 +3,39 @@ import {wrap} from "../core/wrap";
 // Get the complement of a function.
 export const negate = wrap({
     name: "negate",
+    summary: "Get the logical negation of a predicate function.",
+    docs: process.env.NODE_ENV !== "development" ? undefined : {
+        expects: (`
+            The function expects a single predicate function as its input.
+        `),
+        returns: (`
+            The function builds and returns a function returning true for
+            which inputs producing a falsey value when passed to the
+            inputted predicate and returning false for all inputs producing
+            a truthy value when passed to that predicate.
+        `),
+        examples: [
+            "basicUsage"
+        ],
+        related: [
+            "allPass", "anyPass", "nonePass"
+        ],
+    },
```

Comment:
Yeah, we're getting into generics here which is a rabbit hole I don't even want to think about right now. Without going so far as to rewrite everything in [Typescript](https://www.typescriptlang.org/) or [Flow](https://flow.org/), I've seen people hack it and just add a "moduleType" property onto their output object and parse it by checking against that property.

Pretty weak, but it's what we've got. I have a feeling this will come up again in the future...